.github/workflows/kythe.yml

```diff
@@ -0,0 +1,90 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Install Kythe, extract Kzips, and upload to GCP to enable semantic indexing on
+# https://cs.opensource.google/opentitan
+
+name: kythe
+
+on:
+  # For debugging, it's useful to have the action run whenever the PR is
+  # modified. This can be accomplished by setting a bare `pull_request:` with no
+  # items specified.
+  #
+  # Under normal circumstances, the action should only run when PRs are merged,
+  # and possibly less frequently.
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges-1
+  #
+  # TODO(dmcardle) Run only when PRs are merged.
+  pull_request:
+
+jobs:
+  build_and_upload_kzips:
+    name: 'Build and upload kzips for Kythe/Codesearch'
+
+    # Kythe v0.0.59 expects a newer glibc than ubuntu-18.04 and ubuntu-20.04
+    # provide. That leaves ubuntu-22.04 as the next available virtual
+    # environment on GitHub Actions.
+    # https://github.com/actions/virtual-environments#available-environments
+    #
+    # For example, when running on ubuntu-20.04, the Bazel build step will fail
+    # with a message like this:
+    #   /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found
+    #   (required by external/kythe_release/extractors/bazel_cxx_extractor)
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Report glibc version
+        run: |
+          ldd --version
+
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install Kythe
+        run: |
+          set -ex
+          . util/build_consts.sh
+
+          mkdir -p $KYTHE_DIR
+
+          wget --quiet -O /tmp/kythe.tar.gz \
+            "https://github.com/kythe/kythe/releases/download/$KYTHE_VERSION/kythe-$KYTHE_VERSION.tar.gz"
+
+          tar --no-same-owner -xvzf /tmp/kythe.tar.gz --directory ./tools
+
+          echo 'b2a8af00ee551ff35a275c7bf69d095e47cc226d5731e5e724e413b5028707fc  /tmp/kythe.tar.gz' | \
```

Comment:
nit: would it be better to check the hash first before extracting?

Reply:
Yes, absolutely.